images/generic/src/image.rs

```diff
@@ -60,10 +63,16 @@ impl GenericImage {
         Self {
             descriptor: descriptor.into(),
             arguments: vec![],
+            env_vars: HashMap::new(),
             wait_for: WaitFor::Nothing,
         }
     }
 
+    pub fn with_env_var<S: Into<String>>(mut self, key: S, value: S) -> Self {
```

Comment:
I think we should have two parameters here otherwise, you cannot do something like:

`with_env_var("my_var", a_variable)`

if `a_variable` is of type `String` because `S` will already resolve to `&'static str` from the `key` parameter.

Reply:
good catch! Fixed!